pkg/reconciler/trigger/trigger.go

```diff
@@ -207,6 +240,37 @@ func (r *Reconciler) reconcile(ctx context.Context, t *v1alpha1.Trigger) error {
 	return nil
 }
 
+func (r *Reconciler) triggerMarkedReadyByDependency(dependencyObjRef corev1.ObjectReference, ctx context.Context, t *v1alpha1.Trigger) error {
+	gvk := schema.GroupVersionKind{
+		Group:   dependencyObjRef.GroupVersionKind().Group,
+		Version: dependencyObjRef.GroupVersionKind().Version,
+		Kind:    dependencyObjRef.Kind,
+	}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	_, lister, err := r.buildInformerFactory.Get(gvr)
+	if err != nil {
+		logging.FromContext(ctx).Error(fmt.Sprintf("Error getting a lister for a resource with gvk '%+v', gvr '%+v'", gvk, gvr), zap.Error(err))
+		t.Status.MarkDependencyReadyUnknown("FailedToListResource", "Failed to list resource")
+		return err
+	}
+	dependencyObj, err := lister.ByNamespace(dependencyObjRef.Namespace).Get(dependencyObjRef.Name)
```

Comment:
We should probably use the trigger's namespace here (in combination with validation that ensures the annotation's namespace is the same as the trigger's namespace), or always set the dependency's namespace to the same as the trigger's namespace.

Reply:
Totally agree, force in the Trigger's namespace.